crates/mongodb-agent-common/src/mongodb/selection.rs

```diff
@@ -323,6 +323,10 @@ mod tests {
 
         let query_plan = plan_for_query_request(&students_config(), query_request)?;
 
+        // TODO: This selection illustrates that we end up looking up the relationship twice (once
+        // with the key `class_students`, and then with the key `class_students_0`). This is
+        // because the queries on the two relationships have different scope names. The query would
+        // work with just one lookup. Can we do that optimization?
```

Comment:
Can we create a ticket for this TODO as well please.

Reply:
Sure thing, I filed https://hasurahq.atlassian.net/browse/MDB-164